diff --git a/gstlal-ugly/python/svd_bank.py b/gstlal-ugly/python/svd_bank.py
index c7953c9..1f10a99 100755
--- a/gstlal-ugly/python/svd_bank.py
+++ b/gstlal-ugly/python/svd_bank.py
@@ -60,8 +60,12 @@ from gstlal import templates
 #
 
 def read_approximant(xmldoc):
-	approximant=ligolw_process.get_process_params(xmldoc, "tmpltbank", "--approximant")
-	approximant=approximant[0]
+	try:
+		approximant=ligolw_process.get_process_params(xmldoc, "tmpltbank", "--approximant")
+		approximant=approximant[0]
+	except:
+		approximant=ligolw_process.get_process_params(xmldoc, "makeBank", "--approximant")
+		approximant=approximant[0]
 
 	supported_approximants=[u'FindChirpSP', u'TaylorF2', u'IMRPhenomB']
 	if approximant not in supported_approximants:
