diff --git a/gst/audioresample/resample.c b/gst/audioresample/resample.c
index 4410bdd..f631e7d 100644
--- a/gst/audioresample/resample.c
+++ b/gst/audioresample/resample.c
@@ -1174,7 +1174,7 @@ speex_resampler_process_float (SpeexResamplerState * st,
   if (st->magic_samples[channel_index])
     olen -= speex_resampler_magic (st, channel_index, &out, olen);
   if (!st->magic_samples[channel_index]) {
-    while (ilen && olen) {
+    while (ilen) {
       spx_uint32_t ichunk = (ilen > xlen) ? xlen : ilen;
       spx_uint32_t ochunk = olen;
 
@@ -1229,7 +1229,7 @@ speex_resampler_process_int (SpeexResamplerState * st,
 
   st->out_stride = 1;
 
-  while (ilen && olen) {
+  while (ilen) {
     spx_word16_t *y = ystack;
     spx_uint32_t ichunk = (ilen > xlen) ? xlen : ilen;
     spx_uint32_t ochunk = (olen > ylen) ? ylen : olen;
