diff --git a/gstlal/lib/gstlal/gstlal_peakfinder.ct b/gstlal/lib/gstlal/gstlal_peakfinder.ct
index 7a61bef..1b728b2 100644
--- a/gstlal/lib/gstlal/gstlal_peakfinder.ct
+++ b/gstlal/lib/gstlal/gstlal_peakfinder.ct
@@ -19,6 +19,8 @@ int NAME(gstlal,peak_over_window)(struct gstlal_peak_state *state, const TYPE *d
 	guint sample, channel;
 	TYPE *maxdata = MEMBER(state->values.as);
 	guint *maxsample = state->samples;
+	double absdata;
+	double absdatachannel;
 	
 	/* clear the state array */
 	gstlal_peak_state_clear(state);
@@ -26,9 +28,10 @@ int NAME(gstlal,peak_over_window)(struct gstlal_peak_state *state, const TYPE *d
 	/* Find maxima of the data */
 	for(sample = 0; sample < length; sample++) {
 		for(channel = 0; channel < state->channels; channel++) {
-			if(ABSFUNC(*data) > ABSFUNC(maxdata[channel]) && ABSFUNC(*data) > state->thresh) {
+			absdata = ABSFUNC(*data);
+			if(absdata > state->thresh && absdata > (absdatachannel = ABSFUNC(maxdata[channel]))) {
 				/* only increment events if the previous value was 0 */
-				if (ABSFUNC(maxdata[channel]) == 0)
+				if (absdatachannel == 0)
 					state->num_events += 1;
 				maxdata[channel] = *data;
 				maxsample[channel] = sample;
